src/Extension/FluentVersionedExtension.php

```diff
@@ -304,29 +332,99 @@ protected function isLocalisedInStage($stage, $locale = null)
             $table .= self::SUFFIX_LIVE;
         }
 
+        if (isset(static::$idsInLocaleCache[$locale][$table])) {
```

Comment:
I feel like this isset check should include the owner's ID like the line below

Reply:
The reason I didn't is because we know all the IDs that exist in the locale for that table - the absence of an ID implies it's not there. So items that _aren't_ in that stage/locale will skip this cache if you isset check the ID too.